modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -167,6 +219,58 @@
 					this._processFile(imageFile, event.editor);
 				}
 			}
+			else if (event.data && event.data.type === 'html') {
+				const editor = event.editor;
+				const fragment = CKEDITOR.htmlParser.fragment.fromHtml(
+					event.data.dataValue
+				);
+
+				fragment.forEach((element) => {
+					if (
+						element.type === CKEDITOR.NODE_ELEMENT &&
+						element.name === 'img'
+					) {
+						const base64Regex = /^data:image\/(.*);base64,/;
+						const match = element.attributes.src.match(base64Regex);
+
+						if (element.attributes.src && match) {
+							const src = element.attributes.src;
+							const extension = match[1];
+							const name = `${Date.now().toString()}.${extension}`;
+
+							// Mark this "src" attribute as something we want to replace
+
+							this._imageSrcToReplace.push(src);
+
+							this._fetchBlob(src)
+								.then((blob) => {
+									const file = new File([blob], name, {
```

Comment:
> Could we reuse _handleFiles?
I don't think so, but I can have a look

Reply:
Well if I use `_handleFiles` the pasted images gets duplicated, I'll see if I can get around this.

Reply:
> Well if I use _handleFiles the pasted images gets duplicated, I'll see if I can get around this.

Silly me! 

It's kind of obvious why that happens since  the `_processFile` function has 

```javascript
editor.insertElement(el);
```

And I don't want to add an  `if` to get around it.

I also think duplicating this is OK

```javascript
const element = CKEDITOR.dom.element.createFromHtml(
  `<img src="${src}">`
 );

editor.fire('imageAdd', {
  el: element,
  file,
});
```

So I'll keep that for now